src/xcode/ENA/HealthCertificateToolkit/Sources/HealthCertificateToolkit/DigitalGreenCertificateAccess.swift

```diff
@@ -3,13 +3,65 @@
 //
 
 import Foundation
+import base45_swift
 import SwiftCBOR
 import JSONSchema
 
-struct CertificateAccess {
+public typealias Base45 = String
+public typealias CBORData = Data
+
+public struct DigitalGreenCertificateAccess {
+
+    // MARK: - Init
+
+    public init() {}
+
+    // MARK: - Public
+
+    public func extractCBORWebTokenHeader(from base45: Base45) -> Result<CBORWebTokenHeader, CertificateDecodingError> {
+        let cborDataResult = extractCBOR(from: base45)
+
+        switch cborDataResult {
+        case let .success(cborData):
+            return extractHeader(from: cborData)
+        case let .failure(error):
+            return .failure(error)
```

Comment:
We can't log the error here, because we don't have the logging here, right?

Reply:
Right. I added a point to dev camp, to discuss a solution for that.